client/src/components/WaitingRoom/index.js

```diff
@@ -10,64 +10,54 @@ class WaitingRoomBase extends Component {
     this.state = {
       players: [],
       gameId: '',
-      timeLimit: 10,
       started: false
     };
 
-    this.onPlayerListUpdate = this.onPlayerListUpdate.bind(this);
-    this.onGameStart = this.onGameStart.bind(this);
-    this.startGame = this.startGame.bind(this);
+    this.enterGame = this.enterGame.bind(this);
   }
 
-  async componentDidMount() {
-    //Grab gameId & timeLimit to set those as well as gameName
+  componentDidMount() {
     const { id } = this.props.match.params;
-    const gameInstance = await this.props.firebase.game(id).get();
-    this.setState({
-      gameId: id,
-      timeLimit: gameInstance.data().timeLimit,
-      started: gameInstance.data().hasStarted,
-      players: gameInstance.data().players
+    const gameInstance = this.props.firebase.game(id);
+
+    // Listens for changes in the database for player list & status of whether the game has started
+
+    gameInstance.get()
+    .then(docSnapshot => {
+      if(docSnapshot.exists) {
+        gameInstance.onSnapshot((snapshot) => {
```

Comment:
Listeners should be detached in the componentWillUnmount(). See the chat for info!